core/tests/protractor_desktop/profileFeatures.js

```diff
@@ -15,3 +15,178 @@
 /**
  * @fileoverview End-to-end tests for user profile features.
  */
+
+var DEFAULT_BIO = 'This user has not supplied a bio yet.';
+var PLACEHOLDER_INTEREST_TEXT = 'none specified';
+
+var users = require('../protractor_utils/users.js');
+var general = require('../protractor_utils/general.js');
+var waitFor = require('../protractor_utils/waitFor.js');
+
+var ProfilePage = require('../protractor_utils/ProfilePage.js');
+var PreferencesPage = require('../protractor_utils/PreferencesPage.js');
+
+describe('Un-customized profile page for current, logged-in user', function() {
+  var TEST_USERNAME = 'currentDefaultProfileFeatures';
+  var TEST_EMAIL = TEST_USERNAME + '@example.com';
+
+  var profilePage = null;
+
+  beforeAll(function() {
+    profilePage = new ProfilePage.ProfilePage();
+    users.createUser(TEST_EMAIL, TEST_USERNAME);
+  });
+
+  beforeEach(function() {
+    users.login(TEST_EMAIL);
+    profilePage.get(TEST_USERNAME);
+  });
+
+  it('displays profile photo', function() {
```

Comment:
For e2e tests, try to batch the "it"s a little bit, since otherwise the beforeEach and afterEach run each time. We normally keep them separate for unit tests, but since the setup/teardown takes an order of magnitude longer for e2e tests, we sometimes batch the tests together if it doesn't cause problems.

Reply:
So e.g. I think these four "it"s could be condensed into one.

Reply:
Good to know. Thanks for the explanation! Resolved in ddce9cd.